### PR TITLE
Added GiftCardManagement

### PR DESCRIPTION
**Pull request by Domarkaite**
Basic crud for gift cards

### CODE REVIEW

File: Shopipy.Persistence/Models/GiftCard.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+public class GiftCard
+{
+    [Key]
+    public int GiftCardId { get; set; }
+
+    [ForeignKey("Business")]
+    public required int BusinessId { get; set; }
+
+    [ForeignKey("Category")]
```
Comment (piatrashkakanstantinass) [inline Shopipy.Persistence/Models/GiftCard.cs:14]: How categories are used?

Comment (Domarkaite) [reply Shopipy.Persistence/Models/GiftCard.cs:14]: image

Comment (piatrashkakanstantinass) [reply Shopipy.Persistence/Models/GiftCard.cs:14]: how does it help with buying giftcards?

Comment (piatrashkakanstantinass) [reply Shopipy.Persistence/Models/GiftCard.cs:14]: I see the possible point of having a product called giftcard with a bit of special handling

Comment (piatrashkakanstantinass) [reply Shopipy.Persistence/Models/GiftCard.cs:14]: but as in the message, what's the point of overthinking, just don't bother with how giftcards will be bought

Comment (Domarkaite) [reply Shopipy.Persistence/Models/GiftCard.cs]: eh, must've misunderstood. fixed

File: Shopipy.GiftcardManagement/Services/GiftCardService.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
+﻿using Shopipy.Persistence.Models;
+using Shopipy.Persistence.Repositories;
+using Shopipy.Shared.Services;
+
+namespace Shopipy.GiftcardManagement.Services
+{
+    public class GiftCardService(IGenericRepository<GiftCard> _giftCardRepository) : IGiftCardService
+    {
+        public async Task<GiftCard> CreateGiftCardAsync(GiftCard giftCard, int businessId)
+        {
```
Comment (ashkema) [inline Shopipy.GiftcardManagement/Services/GiftCardService.cs:10]: Add data validation:
- Check that `ValidUntil` is earlier than `ValidFrom`
- Check that `AmountOriginal`  is a positive value

File: Shopipy.GiftcardManagement/Controllers/GiftCardController.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        var giftCards = await _giftCardService.GetAllGiftCardsOfBusinessAsync(businessId, top, skip);
+        var count = await _giftCardService.GetGiftCardCountAsync(businessId);
+
+        return Ok(new PaginationResultDto<GiftCardResponseDTO>
+        {
+            Data = giftCards.Select(_mapper.Map<GiftCardResponseDTO>),
+            Count = count
+        });
+    }
+
```
Comment (ashkema) [inline Shopipy.GiftcardManagement/Controllers/GiftCardController.cs:41]: Add batch creation for gift cards (e.g. use case: a business owner wants to create 100 gift cards to distribute as a promotion)

**Review by ashkema — CHANGES_REQUESTED** [review]

File: Shopipy.GiftcardManagement/Controllers/GiftCardController.cs (hunk 3 above)
Comment (piatrashkakanstantinass) [reply Shopipy.GiftcardManagement/Controllers/GiftCardController.cs:41]: Don’t create new requirements

**Comment by piatrashkakanstantinass** [comment]
Make sure to add new business access policy to controller

File: Shopipy.GiftcardManagement/Controllers/GiftCardController.cs (hunk 3 above)
Comment (Domarkaite) [reply Shopipy.GiftcardManagement/Controllers/GiftCardController.cs:41]: > Don’t create new requirements

agreed

File: Shopipy.GiftcardManagement/Services/GiftCardService.cs (hunk 2 above)
Comment (Domarkaite) [reply Shopipy.GiftcardManagement/Services/GiftCardService.cs:10]: done i'd say

**Review by ashkema — APPROVED** [review]

**Pull request merged**